spec/unit/manifest_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'etc'
+
+module ThreeScale
+  module Backend
+    describe Manifest do
+      describe '.thread_safe?' do
+
+        it 'returns false' do
+          expect(described_class.thread_safe?).to eq(false)
+        end
+
+      end
+
+      describe '.compute_workers' do
+
+        before :each do
+          ENV::delete('PUMA_WORKERS')
```

Comment:
Now about how to test this - it is not correct to modify the current process' environment as it modifies global state for all the tests. What you need is to mock the retrieval of `"PUMA_WORKERS"` from the `ENV` object. This should be done by mocking `ENV#[]` when the argument is `"PUMA_WORKERS"`, something similar to:
```ruby
allow(ENV).to receive(:[]).with("PUMA_WORKERS").and_return(nil)
```

Reply:
@unleashed I'm not really convinced mocking is the right way. Because then you are really coupling it to the implementation - which could use any of the dozen ways to get the env `.fetch`, `.to_h`, `assoc`, `each`, `values_at`, ...

Standard way is to have global around filter that does `env = ENV.to_h; yield; ENV.replace(env)`

Reply:
You'd need to place the `replace` in an `ensure` statement and call it with a block - the whole thing looks rather unergonomic. You can also make a double out of a `ENV` and ensure the specific key is not there - in fact there is `stub_const("ENV", env_value)` which seems to be designed specifically for this https://relishapp.com/rspec/rspec-mocks/v/3-7/docs/mutating-constants/stub-defined-constant